src/StorageRent.sol

```diff
@@ -0,0 +1,578 @@
+            emit Rent(msg.sender, fids[i], units);
+        }
+    }
+
+    /**
+     * @notice Force refresh the cached Chainlink ETH/USD price.
+     */
+    function refreshPrice() external {
+        if (!hasRole(ADMIN_ROLE, msg.sender) && !hasRole(TREASURER_ROLE, msg.sender)) revert Unauthorized();
+        _refreshPrice();
+    }
+
+    /**
+     * @notice Change the USD price per storage unit.
+     *
+     * @param usdPrice The new unit price in USD. Fixed point value with 8 decimals.
+     */
+    function setPrice(uint256 usdPrice) external {
+        if (!hasRole(ADMIN_ROLE, msg.sender) && !hasRole(TREASURER_ROLE, msg.sender)) revert Unauthorized();
+        emit SetPrice(usdUnitPrice, usdPrice);
+        usdUnitPrice = usdPrice;
+    }
+
+    /**
+     * @notice Change the maximum supply of storage units.
+     *
+     * @param max The new maximum supply of storage units.
+     */
+    function setMaxUnits(uint256 max) external onlyAdmin {
+        /* 1.6e7 = 64TB @ 4MB/unit */
+        if (max > 1.6e7) revert InvalidMaxUnits();
+        emit SetMaxUnits(maxUnits, max);
+        maxUnits = max;
+    }
+
+    /**
+     * @notice Change the deprecationTimestamp.
+     *
+     * @param timestamp The new deprecationTimestamp.
+     */
+    function setDeprecationTimestamp(uint256 timestamp) external onlyAdmin {
+        if (timestamp < block.timestamp) revert InvalidDeprecationTimestamp();
+        emit SetDeprecationTimestamp(deprecationTimestamp, timestamp);
+        deprecationTimestamp = timestamp;
+    }
+
+    /**
+     * @notice Change the priceFeedCacheDuration.
+     *
+     * @param duration The new priceFeedCacheDuration.
+     */
+    function setCacheDuration(uint256 duration) external onlyAdmin {
+        emit SetCacheDuration(priceFeedCacheDuration, duration);
+        priceFeedCacheDuration = duration;
+    }
+
+    /**
+     * @notice Change the uptimeFeedGracePeriod.
+     *
+     * @param period The new uptimeFeedGracePeriod.
+     */
+    function setGracePeriod(uint256 period) external onlyAdmin {
+        emit SetGracePeriod(uptimeFeedGracePeriod, period);
+        uptimeFeedGracePeriod = period;
+    }
+
+    /**
+     * @notice Set the vault address that can receive funds from this contract.
+     *
+     * @param vaultAddr The new vault address.
+     */
+    function setVault(address vaultAddr) external onlyAdmin {
+        emit SetVault(vault, vaultAddr);
+        vault = vaultAddr;
+    }
+
+    /**
+     * @notice Withdraw a specified amount of ether from the contract balance to the vault.
+     *
+     * @param amount The amount of ether to withdraw.
+     */
+    function withdraw(uint256 amount) external onlyTreasurer {
+        emit Withdraw(vault, amount);
+        _sendNative(vault, amount);
+    }
+
+    /**
+     * @notice Native token transfer helper.
+     */
+    function _sendNative(address to, uint256 amount) internal {
+        if (address(this).balance < amount) revert InsufficientFunds();
```

Comment:
Just clarity and consistency—this was the same transfer code that was in the `NameRegistry`, and itself is close to [what OZ does](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/e50c24f5839db17f46991478384bfda14acfb830/contracts/utils/Address.sol#L65).

If we omit this, it will just revert with `CallFailed` instead.

Reply:
I guess this does have a gas impact in some cases too, since it avoids trying to make the external call.